specs/interop.md

```diff
@@ -0,0 +1,540 @@
+      - [Executing Messages](#executing-messages)
+    - [L2ToL2CrossDomainMessenger](#l2tol2crossdomainmessenger)
+      - [Transferring Ether in a Cross Chain Message](#transferring-ether-in-a-cross-chain-message)
+      - [Interfaces](#interfaces)
+        - [Sending Messages](#sending-messages)
+        - [Executing Messages](#executing-messages-1)
+    - [L1Block](#l1block)
+  - [SystemConfig](#systemconfig)
+    - [`DEPENDENCY_SET` UpdateType](#dependency_set-updatetype)
+  - [L1Attributes](#l1attributes)
+  - [Fault Proof](#fault-proof)
+  - [Sequencer Policy](#sequencer-policy)
+  - [Block Building](#block-building)
+  - [Sponsorship](#sponsorship)
+  - [Security Considerations](#security-considerations)
+    - [Forced Inclusion of Cross Chain Messages](#forced-inclusion-of-cross-chain-messages)
+    - [Cross Chain Message Latency](#cross-chain-message-latency)
+    - [Dynamic Size of L1 Attributes Transaction](#dynamic-size-of-l1-attributes-transaction)
+    - [Maximum Size of the Dependency Set](#maximum-size-of-the-dependency-set)
+    - [Mempool](#mempool)
+    - [Reliance on History](#reliance-on-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Interoperability
+
+The ability for a blockchain to easily read the state of another blockchain is called interoperability.
+Low latency interoperability allows for horizontally scalable blockchains, a key feature of the superchain.
+
+Note: this document references an "interop network upgrade" as a temporary name. The actual name of the
+network upgrade will be included in this document in the future.
+
+## Overview
+
+| Term          | Definition                                               |
+|---------------|----------------------------------------------------------|
+| Source Chain   | A blockchain that includes an initiating message |
+| Destination Chain  | A blockchain that includes an executing message |
+| Initiating Message | A transaction submitted to a source chain that authorizes execution on a destination chain |
+| Executing Message | A transaction submitted to a destination chain that corresponds to an initiating message |
+| Cross Chain Message | The cumulative execution and side effects of the initiating message and executing message |
+| Dependency Set | The set of chains that originate initiating transactions where the executing transactions are valid |
+
+A total of two transactions are required to complete a cross chain message. The first transaction is submitted
+to the source chain which authorizes execution on a destination chain. The second transaction is submitted to a
+destination chain, where the block builder SHOULD only include it if they are certain that the first transaction was
+included in the source chain.
+
+The term "block builder" is used interchangeably with the term "sequencer" for the purposes of this document but they need not be the same
+entity in practice.
+
+## The Dependency Set
```

Comment:
If part of the reason for the dependency set is to reduce the chance of a dos, there's still a possibility that it _could_ happen even with a limited/trusted dependency set. What happens in this scenario? How can you recover from it? Would the chain operator need to update the L1 config to remove the spamming chain from the dependency set? Would it be possible (although maybe not ideal) for the sequencer just to ignore the spam?

I don't think interop messages have a higher priority of being included in a block over non-interop txs, right? So a dos would mean a degraded performance for inclusion of every tx?

Reply:
> there's still a possibility that it could happen even with a limited/trusted dependency set

Can you explain how? The executing message isn't force included and pays for gas.

The denial of service comes from looking for data that doesn't exist

The interop executing messages do not have a higher priority for being included over regular messages.